cmd/virsnap/clean.go

```diff
@@ -120,27 +126,34 @@ func cleanRun(cmd *cobra.Command, args []string) {
 				}
 
 				if accepted {
-					log.Infof("Removing snapshot \"%s\" of VM \"%s\".",
-						snapshots[i].Descriptor.Name, vm.Descriptor.Name)
+					logger.Infof("removing snapshot '%s' of VM '%s'.",
+						snapshots[i].Descriptor.Name,
+						vm.Descriptor.Name,
+					)
 
 					err = snapshots[i].Instance.Delete(0)
 					if err != nil {
-						log.Errorf("Could not remove snapshot \"%s\" of VM \"%s\". "+
-							"Skipping this VM: %v", snapshots[i].Descriptor.Name,
-							vm.Descriptor.Name, err)
+						logger.Errorf("skipping VM '%s': error, unable to remove snapshot '%s' of VM '%s': %s",
+							vm.Descriptor.Name,
+							snapshots[i].Descriptor.Name,
+							err,
+						)
 						failed = true
 						return // we are in an anonymous function
 					}
 				} else {
-					log.Infof("Skipping removal of snapshot \"%s\" of VM \"%s\".",
-						snapshots[i].Descriptor.Name, vm.Descriptor.Name)
+					logger.Infof("skipping removal of snapshot '%s' of VM '%s'",
+						snapshots[i].Descriptor.Name,
+						vm.Descriptor.Name,
+					)
 				}
 			}
 		}()
 	}
-
+	// TODO (obitech): improve error handling
+	// See: https://blog.golang.org/errors-are-values
```

Comment:
You're absolutely right! The thing is, there could potentially be many errors, since we do not skip the creation of snapshots of another VM if a previous one failed. I thought about an slice of errors or another "collection type" but was not happy either. Your idiomatic golang solution is highly appreciated!

Reply:
I just wanted to add the `TODO` here so we have it on our map. I would propose to discuss the changes in an issue or another PR?

Reply:
I agree with you.